### PR TITLE
Feature/JC19-373-search-anime

### PR DESCRIPTION
**Pull request by BKrasnov**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_API_KEY=3df19916-03c9-47de-ab5c-5619376c2cef
+VITE_API_URL=https://api.camp-js.saritasa.rocks/api/v1
```
Comment (artem-sotnichenko-saritasa) [inline .env:2]: Please remove .env file from repository

File: apps/vanilla/src/scripts/pagination.ts [hunk 2]
```diff
@@ -26,20 +26,28 @@ export class Pagination {
...
       this.paginationContainer.append(this.createButton({ content: '&#9668;', value: 'previous_page' }));
     }
     if (this.startPage < numberOfDisplayedPages) {
       for (let i = 1; i <= numberOfDisplayedPages; i++) {
-        this.paginationContainer.append(this.createButton({ content: i }));
+        if (i <= this.pagesCount) {
+          this.paginationContainer.append(this.createButton({ content: i }));
+        } else {
+          break;
+        }
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/scripts/pagination.ts:45]:
```suggestion
        if (i > this.pagesCount) {
          break;
        }
        
        this.paginationContainer.append(this.createButton({ content: i }));
```

File: apps/vanilla/src/scripts/search.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Initialize select.
+ * @param selectContainer The block where the search is located.
+ * @param sendSearch Send the search order.
+ */
+export function initializeSearch(
+  selectContainer: HTMLSelectElement,
+  sendSearch: Function,
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/scripts/search.ts:8]: Specify function type as narrow as possible. In your case, it would be `(searchText: string) => void`

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

File: apps/vanilla/src/scripts/search.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Initialize select.
+ * @param selectContainer The block where the search is located.
+ * @param sendSearch Send the search order.
+ */
+export function initializeSearch(
+  selectContainer: HTMLSelectElement,
+  sendSearch: (searchText: string) => void,
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/scripts/search.ts:8]: You can declare type and reuse it in `addSearch` function

**Pull request closed without merging**